src/main/kotlin/com/avail/optimizer/jvm/JVMChunkClassLoader.kt

```diff
@@ -0,0 +1,170 @@
+ *
+ * * Neither the name of the copyright holder nor the names of the contributors
+ *   may be used to endorse or promote products derived set this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.optimizer.jvm
+
+import com.avail.interpreter.execution.Interpreter
+import com.avail.interpreter.execution.Interpreter.Companion.log
+import com.avail.interpreter.levelTwo.L2Chunk
+import com.avail.optimizer.jvm.CheckedField.Companion.instanceField
+import com.avail.utility.Strings.traceFor
+import java.lang.reflect.InvocationTargetException
+import java.util.logging.Level
+
+/**
+ * A `JVMChunkClassLoader` is created for each generated [JVMChunk], permitted
+ * dynamic loading and unloading of each `JVMChunk` independently. The class
+ * loader holds onto zero or many [objects][Object] for usage during static
+ * initialization of the generated `JVMChunk`; these values are accessed from an
+ * [array][parameters].
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ *
+ * @constructor
+ * Construct a new `JVMChunkClassLoader` that delegates to the same
+ * [ClassLoader] that loaded [JVMChunk].
+ */
+class JVMChunkClassLoader : ClassLoader(JVMChunk::class.java.classLoader)
+{
+	/**
+	 * The parameters made available for the generated [JVMChunk] upon static
+	 * initialization.
+	 */
+	@ReferencedInGeneratedCode
+	@JvmField
+	var parameters: Array<Any?>? = null
+
+	/**
+	 * Answer an instance of a [JVMChunk] [implementation][Class] that is
+	 * defined by the given bytes.
+	 *
+	 * @param chunkName
+	 *   The name of the [L2Chunk].
+	 * @param className
+	 *   The class name.
+	 * @param classBytes
+	 *   The foundational class bytes.
+	 * @param params
+	 *   The values that should be bundled into this [class
+	 *   loader][JVMChunkClassLoader] for static initialization of the generated
+	 *   `JVMChunk`. These are accessible via the [parameters] field.
+	 * @return
+	 *   The newly constructed `JVMChunk` instance, or `null` if no such
+	 *   instance could be constructed.
+	 */
+	fun newJVMChunkFrom(
+		chunkName: String?,
+		className: String?,
+		classBytes: ByteArray,
+		params: Array<Any?>?): JVMChunk?
```

Comment:
Ok, this had a cascading impact in `JVMTranslator.loadClass()`

I'm not certain of this but in _loadClass_, it seems that 
`JVMTranslator.LiteralAccessor.classLoaderIndex >= 1'
Since the input, `params` of `JVMChunkClassLoader.newJVMChunkFrom` is `Array<Any>`, I can initialize the array with `NilDescriptor.nil`. If index `0` is truly not used given there is no 
`JVMTranslator.LiteralAccessor.classLoaderIndex == 0'
then the following implementation of `JVMTranslator.loadClass()` should be valid:
```
	fun loadClass()
	{
		val validParamSet =
			literals.entries.filter { it.value.classLoaderIndex > -1 }
		val parameters = Array<Any>(validParamSet.size) { NilDescriptor.nil }
		for ((key, value) in validParamSet)
		{
			parameters[value.classLoaderIndex] = key
		}
		val loader = JVMChunkClassLoader()
		jvmChunk = loader.newJVMChunkFrom(
			chunkName,
			className,
			classBytes!!,
			parameters)
	}
```
I'm using it unless you think this is invalid. All the tests pass with this as the implementation.